src/js/utils/Select.jsx

```diff
@@ -115,9 +133,26 @@ class Select extends Component {
     }
 
     if (showLabelTooltip) {
-      const valueMapped = multi && value ?
-        this.props.value.map(v => v?.[labelKey] ?? v?.label) : [];
-      const valueLabel = multi ? valueMapped.join(', ') : (value.label || value.name);
+      let valueLabel = '';
+      if (value) {
+        if (multi) {
+          valueLabel = this.props.value
+            .map((v) => {
+              const label = v?.[labelKey] ?? v?.label;
+              if (label) {
+                return label;
+              }
+              // if there are no labels on value item
+              // then try to extract these labels from select options
+              const option = this.options.find(it => it?.id && (it?.id === v?.id));
+              return option?.[labelKey] ?? option.label;
+            })
+            .join(', ');
+        } else {
```

Comment:
I prefer doing `return` in `if` statement instead of `else`, but it's just my personal preference